src/MuTalk-Tests/MTRandomClassMutantSelectionStrategyTest.class.st

```diff
@@ -28,7 +28,14 @@ MTRandomClassMutantSelectionStrategyTest >> testAround33PercentOfMutantsForEachC
 			            (MTMockRandomGroupedMutantSelectionStrategy
 				             for: MTRandomClassMutantSelectionStrategy new
 				             andSeed: seed).
-	mutations := analysis generateMutations truncateTo: 100.
-	mutationsDic := (mutations groupedBy: #originalClass) collect: #size.
-	self assert: mutationsDic values equals: expected
+
+	mutations := analysis generateMutations truncateTo: maximum.
+	mutationsDic := (mutations groupedBy: #originalClass) collect: [
+		                :mutationList | "Here we divide the number of mutants for each class by the total of mutants to obtain a percentage, and we check if it's close to the theoretical 1/3 with a precision of 0.1, because the small number of mutants can induce a quite large gap"
+		                mutationList size / maximum asFloat
+			                closeTo: 1 / 3 asFloat
+			                precision: 0.1 ].
+
+	self assert:
+		(mutationsDic values fold: [ :bool1 :bool2 | bool1 and: bool2 ])
```

Comment:
So you are testing that the results are evenly distributed by the grouping condition?
It's a fun way to test it ^^.

I understand you did this because the default random generator changed in Pharo11 or Pharo12, making tests not compatible, right?

Alternative solution: have a set of tests for Pharo<11 and a set of tests for Pharo>=11

Also, I see that Pharo9 is breaking because `truncateTo:` is not defined in collections.
Maybe we can (should?) drop Pharo9 support?

Reply:
At first I checked that the results matched exactly the expected distribution (like #( 30 31 39 ) ), but yes in previous pharo versions the expected distributions are not the same because, even with the same seed, the random generator don't give the same results.

Yes I guess it could be a good (maybe even better) solution to have a set for pharo >= 12 and a set for pharo <= 11.

Given the fact that pharo 9 is in deprecated distributions in the pharo launcher, I think it's acceptable to drop pharo 9 support.

Reply:
But I don't know how to make tests for specific versions of Pharo :') But I guess it's not that hard